packages/client/src/createTRPCClientProxy.ts

```diff
@@ -66,9 +68,12 @@ type DecorateProcedure<TProcedure extends AnyProcedure> =
 /**
  * @internal
  */
-type DecoratedProcedureRecord<TProcedures extends ProcedureRouterRecord> = {
+type DecoratedProcedureRecord<
+  TRouter extends AnyRouter,
```

Comment:
We seem to need to be able to keep track of the original router within the generics in order to unwrap it later